packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -671,7 +671,18 @@ export class Function extends FunctionBase {
    * in the same account and region as the stack you are importing it into.
    */
   public static fromFunctionArn(scope: Construct, id: string, functionArn: string): IFunction {
-    return Function.fromFunctionAttributes(scope, id, { functionArn });
+    /**
+     * First, validate the functionArn string. If it's ARN with trailing version or alias,
+     * we trim off the trailing to retrieve the real functionArn.
```

Comment:
```suggestion
     * If the functionArn has a trailing version or alias (more than 7 parts when split by ":",
     * we trim off the trailing version/alias to retrieve the real functionArn.
```
Let me know if you think this makes more sense. I don't think we need to say that we're validating the functionArn string.